geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/LuceneIndexCreationProfile.java

```diff
@@ -36,18 +38,25 @@
 
   private Map<String, String> fieldAnalyzers;
 
+  private String serializerClass;
+
   private String regionPath;
 
   /* Used by DataSerializer */
   public LuceneIndexCreationProfile() {}
 
   public LuceneIndexCreationProfile(String indexName, String regionPath, String[] fieldNames,
-      Analyzer analyzer, Map<String, Analyzer> fieldAnalyzers) {
+      Analyzer analyzer, Map<String, Analyzer> fieldAnalyzers, LuceneSerializer serializer) {
     this.indexName = indexName;
     this.regionPath = regionPath;
     this.fieldNames = fieldNames;
     this.analyzerClass = analyzer.getClass().getSimpleName();
     initializeFieldAnalyzers(fieldAnalyzers);
+    if (serializer == null) {
+      this.serializerClass = HeterogeneousLuceneSerializer.class.getSimpleName();
+    } else {
+      this.serializerClass = serializer.getClass().getSimpleName();
```

Comment:
Don't we want the full name and not the simple name?  What happens if there is a name collision...

Reply:
It follows the same idea of handling Analyzer class.